wallet/wallet.go

```diff
@@ -363,12 +363,11 @@ func (w *Wallet) syncWithChain(birthdayStamp *waddrmgr.BlockStamp) error {
 		if err != nil {
 			return err
 		}
-		startHeight := bestHeight - waddrmgr.MaxReorgDepth + 1
+		startHeight := birthdayStamp.Height
 		if startHeight < 0 {
 			startHeight = 0
-		}
-		if birthdayStamp.Height < startHeight {
-			startHeight = birthdayStamp.Height
+		} else if bestHeight < startHeight {
```

Comment:
This is not possible as it's assumed that the birthday is at most the best height in the chain, so this check is not needed.